common/transport/snmp/src/main/java/org/thingsboard/server/transport/snmp/SnmpTransportContext.java

```diff
@@ -127,7 +208,92 @@ public ExecutorService getSnmpCallbackExecutor() {
                 .collect(Collectors.toList());
     }
 
-    //TODO: Extract SNMP methods to enum
+    @EventListener(DeviceUpdatedEvent.class)
+    public void onDeviceUpdatedOrCreated(DeviceUpdatedEvent deviceUpdatedEvent) {
+        Device device = deviceUpdatedEvent.getDevice();
+        DeviceTransportType transportType = Optional.ofNullable(device.getDeviceData().getTransportConfiguration())
+                .map(DeviceTransportConfiguration::getType)
+                .orElse(null);
+        if (!allSnmpDevicesIds.contains(device.getId())) {
+            if (transportType != DeviceTransportType.SNMP) {
+                return;
+            }
+            allSnmpDevicesIds.add(device.getId());
+            if (balancingService.isManagedByCurrentTransport(device.getId().getId())) {
+                establishDeviceSession(device);
+            }
+        } else {
+            if (balancingService.isManagedByCurrentTransport(device.getId().getId())) {
+                DeviceSessionContext sessionContext = sessions.get(device.getId());
+                if (transportType == DeviceTransportType.SNMP) {
+                    if (sessionContext != null) {
+                        updateDeviceSession(sessionContext, device, deviceProfileCache.get(device.getDeviceProfileId()));
+                    } else {
+                        establishDeviceSession(device);
+                    }
+                } else {
+                    destroyDeviceSession(sessionContext);
+                }
+            }
+        }
+    }
+
+    public void onDeviceDeleted(DeviceSessionContext sessionContext) {
+        destroyDeviceSession(sessionContext);
+    }
+
+    public void onDeviceProfileUpdated(DeviceProfile deviceProfile, DeviceSessionContext sessionContext) {
+        updateDeviceSession(sessionContext, sessionContext.getDevice(), deviceProfile);
+    }
+
+    @EventListener(SnmpTransportListChangedEvent.class)
+    public void onSnmpTransportListChanged() {
```

Comment:
Please use TbApplicationEventListener to deduplicate the events. This will minimize recalculations.